source/rust_verify/src/rust_to_vir_expr.rs

```diff
@@ -958,6 +960,16 @@ pub(crate) fn expr_to_vir_with_adjustments<'tcx>(
                 "dereferencing a pointer (here the dereference is implicit)"
             )
         }
+        Adjust::Pointer(PointerCast::Unsize) => {
+            // REVIEW Should we track the size of the array as a fact about the resulting slice?
+            expr_to_vir_with_adjustments(
+                bctx,
+                expr,
+                current_modifier,
+                adjustments,
+                adjustment_idx - 1,
+            )
+        }
```

Comment:
@tjhance does this look right to you?

`Unsize` is "Unsize a pointer/reference value, e.g., `&[T; n]` to `&[T]`."

Reply:
yup